### PR TITLE
fix: Subgraph CI tests

### PR DESCRIPTION
**Pull request by georgi-l95**
**Description**:

Currently the subgraph CI Test is failing due to the events not indexed on the same order that they are expected, however the events and it's details are there.

This PR modifies the queries in order to 
* Fixes a bug where the `HTSFT` events are wrongly indexed due to wrong block number being passed (now changed to `8` which fixes the problem)
* Adds a `prepare-test.sh` script that runs the necessary preparations to execute the subgraph tests
* Adds typing to query results and custom assertions that check each field instead of comparing JSON strings

**Related issue(s)**:

Fixes [#2198](https://github.com/hashgraph/hedera-json-rpc-relay/issues/2198)

**Notes for reviewer**:

image

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by georgi-l95 — COMMENTED** [review]
LG, I like the strategy pattern idea. 👍

**Review by konstantinabl — DISMISSED** [review]

File: tools/subgraph-example/prepare-test.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+echo "graph-local"
+npm run graph-local -- --detach
+sleep 10
+
+echo "create-local"
+npm run create-local
+sleep 1
+
+echo "deploy-local"
+npm run deploy-local -- --network local
```
Comment (AlfredoG87) [inline tools/subgraph-example/prepare-test.sh:29]: blank line missing  EOF

File: tools/subgraph-example/subgraph/networks.json [hunk 2]
```diff
@@ -2,23 +2,23 @@
...
       "startBlock": 10
     },
     "ExampleHTSFT": {
       "address": "0x0000000000000000000000000000000000000409",
-      "startBlock": 10
+      "startBlock": 8
     }
   }
-}
+}
```
Comment (AlfredoG87) [inline tools/subgraph-example/subgraph/networks.json:24]: blank line missing  EOF

**Review by AlfredoG87 — APPROVED** [review]
LGTM, only nits for blank line missing  EOF

**Review by AlfredoG87 — APPROVED** [review]
Nice Work! 👍

**Pull request merged**